### DISCUSSION TITLE
Create Java SDK for Langfuse

### DISCUSSION THREAD

**Discussion by OrGivati** [langfuse/langfuse#4589] [Ideas]
### Describe the feature or potential improvement

Java is still a very popular language(see [last stackOverFlow 2024 report](https://survey.stackoverflow.co/2024/technology#most-popular-technologies-language). developing Java SDK can help many different applications benefit from Langfuse to create better LLM applications in the java ecosystem. 

### Additional information

Integrating Langfuse to [Spring AI](https://docs.spring.io/spring-ai/reference/observability/index.html) will help integrate it more easily into existing applications.

**Comment by marcklingen** [comment]
Thanks for sharing! Let's track interest for this here.

I know a couple of teams who have created typed clients for Java/Go based on the openapi spec: https://cloud.langfuse.com/generated/api/openapi.yml

Have you tried this already?

**Comment by tomAgnicio** [comment]
@marcklingen @OrGivati 
As a java developer I am trying to look for projects that support java for tracing, evaluation, monitoring, ... of RAG/LLMs

The options are very limited today. Any decent integration would be very useful.

**Comment by marcklingen** [comment]
We plan to add support for Java applications via OTel, this is coming soon, tracking this here (please upvote/subscribe): https://github.com/orgs/langfuse/discussions/2509

**Comment by kpavlov** [comment]
There is something to start with or continue: https://github.com/kpavlov/langfuse-jvm

**Comment by marcklingen** [comment]
@kpavlov now that there is an OTel collector, it could be interesting to just use the OTel Java SDK and export traces to langfuse
https://langfuse.com/docs/opentelemetry/get-started

PS: I thought about putting together a simple example here

**Reply by kpavlov** [reply]
I am interested in prompt management usecase

**Comment by marcklingen** [comment]
@Steffen911 released this recently, feedback appreciated!
https://github.com/langfuse/langfuse-java

**Reply by marcklingen** [reply]
+ how to use OTel to trace Spring AI with Langfuse: https://langfuse.com/docs/integrations/spring-ai

**Reply by Steffen911** [reply]
And there's also a couple of examples on how to instrument Quarkus or Spring in Java: https://github.com/langfuse/langfuse-examples/tree/main/applications.

**Comment by fpaupier** [comment]
Thanks for this port @marcklingen and @Steffen911! Actually looking into it for a support with java spring, having some getting started example definitely help (how to push a trace and events), your python doc and wrapper are so good that getting back to a manual API call with careful payload creation is a bit challenging.

**Reply by Steffen911** [reply]
@fpaupier Thank you! 🙌 

Overall, we would not recommend to use the new Java SDK for tracing. Here, the OpenTelemetry integration is significantly better in terms of performance and out-of-the-box functionality. You can check out https://github.com/langfuse/langfuse-examples/tree/main/applications/spring-ai-demo for an example on how to do it with SpringAI.

If you have other use cases, e.g. prompt management, the Java SDK could help - let us know in case you need some examples there.
